src/transaction.rs

```diff
@@ -205,11 +239,93 @@ pub fn create_transaction_internal(
     network: Network,
 ) -> Result<JSTransaction, Box<dyn Error>> {
     let mut builder = Builder::new(network, block_height);
+    let (nullifiers, change) = match inputs {
+        Either::Left(notes) => choose_notes(&mut builder, &notes, extsk, amount)?,
+        Either::Right(utxos) => choose_utxos(&mut builder, &utxos, amount)?,
+    };
+
+    let amount = Amount::from_u64(amount).map_err(|_| "Invalid Amount")?;
+
+    let change_address =
+        decode_payment_address(network.hrp_sapling_payment_address(), change_address)
+            .map_err(|_| "Failed to decode change address")?;
+    if to_address.starts_with(network.hrp_sapling_payment_address()) {
+        let to_address = decode_payment_address(network.hrp_sapling_payment_address(), to_address)
+            .map_err(|_| "Failed to decode sending address")?;
+        builder
+            .add_sapling_output(None, to_address, amount, MemoBytes::empty())
+            .map_err(|_| "Failed to add output")?;
+    } else {
+        let to_address = decode_transparent_address(
+            &network.b58_pubkey_address_prefix(),
+            &network.b58_script_address_prefix(),
+            to_address,
+        )?
+        .ok_or("Failed to decode transparent address")?;
+        builder
+            .add_transparent_output(&to_address, amount)
+            .map_err(|_| "Failed to add output")?;
+    }
+
+    builder
+        .add_sapling_output(None, change_address, change, MemoBytes::empty())
+        .map_err(|_| "Failed to add change")?;
+
+    prove_transaction(builder, nullifiers)
+}
 
+fn choose_utxos(
+    builder: &mut Builder<Network, OsRng>,
+    utxos: &[Utxo],
+    amount: u64,
+) -> Result<(Vec<String>, Amount), Box<dyn Error>> {
+    let fee = 2365000u64;
```

Comment:
as I said in the last PR this hardcoded fee will create problems, but if you want we can fix in another PR

Reply:
Will fix in another PR